### PR TITLE
[Feature] Tensorclass device

### PR DESCRIPTION
**Pull request by tcbegley**
## Description

This PR allows the user to specify `device` when instantiating a TensorClass (`@tensorclass` decorated dataclass).

### CODE REVIEW

File: tensordict/prototype/tensorclass.py [hunk 1]
```diff
@@ -102,7 +102,9 @@ def tensorclass(cls: T) -> T:
         fields=[("batch_size", torch.Size, field(default_factory=list))],
     )
 
-    EXPECTED_KEYS = set(datacls.__dataclass_fields__.keys())
+    EXPECTED_KEYS = {
+        k for k in datacls.__dataclass_fields__.keys() if k != "batch_size"
+    }
```
Comment (tcbegley) [inline tensordict/prototype/tensorclass.py:107]: If we don't exclude `batch_size` here then assinging to `.batch_size` attribute attempts to place the supplied value in the underlying tensordict under the key `"batch_size"`.

**Review by tcbegley — COMMENTED** [review]
Because of the logic in `__setattr__`, it's not currently possible to set either `device` or `batch_size` by assigning to those attributes. I understand why we want to prevent setting of arbitrary attributes, but do we want to make an exemption for these two?

**Review by vmoens — APPROVED** [review]
LGTM

**Comment by vmoens** [comment]
> Because of the logic in `__setattr__`, it's not currently possible to set either `device` or `batch_size` by assigning to those attributes. I understand why we want to prevent setting of arbitrary attributes, but do we want to make an exemption for these two?

Yes! But for device you'd need to call .to no?

**Pull request merged**